cpu/stm32/periph/eth.c

```diff
@@ -116,16 +115,16 @@ static uint8_t _link_state = LINK_STATE_DOWN;
  * @return  The value of the MII register accessed. (This should be equal to
  *          @p value, if @p write was `true`.)
  */
-static uint16_t _mii_reg_transfer(unsigned addr, unsigned reg, uint16_t value,
-                                  bool write)
+static uint16_t _mii_reg_transfer(unsigned reg, uint16_t value, bool write)
 {
     unsigned tmp;
+    const uint16_t phy_addr = eth_config.phy_addr;
 
     while (ETH->MACMIIAR & ETH_MACMIIAR_MB) {}
-    DEBUG("[stm32_eth] rw_phy %x (%x): %x\n", addr, reg, value);
+    DEBUG("[stm32_eth] rw_phy %x (%x): %x\n", (unsigned)phy_addr, reg, value);
```

Comment:
```suggestion
    DEBUG("[stm32_eth] rw_phy %x (%x): %x\n", phy_addr, reg, value);
```

you can drop the cast

Reply:
`%x` expects `unsigned int` as type, `phy_addr` is `uint16_t`. (I am aware that for variadic functions integers smaller than `int` get promoted to `int`. But still the correct format specifier would technically be `%hx` and depending on compiler settings, you can get a warning about incorrect format specifier without the cast. I could also use `%hx` instead, but that would be even more obscure than the cast IMO.)

Reply:
I thought the compiler would only complain if the format specifier might truncate the value, not when it's 'too big'.

Reply:
Note that variadic arguments have to go through the stack in any case, as they cannot easily be mapped to registers. E.g. using `%ld` on an `int` on a 64 bit machine is going to grab 32 bits more from the stack than it should.